pkg/k8s/informers/informers.go

```diff
@@ -86,12 +86,16 @@ func WithSMIClients(smiTrafficSplitClient smiTrafficSplitClient.Interface, smiTr
 }
 
 // WithConfigClient sets the config client for the InformerCollection
-func WithConfigClient(configClient configClientset.Interface) InformerCollectionOption {
+func WithConfigClient(configClient configClientset.Interface, meshConfigName, osmNamespace string) InformerCollectionOption {
 	return func(ic *InformerCollection) {
-		informerFactory := configInformers.NewSharedInformerFactory(configClient, DefaultKubeEventResyncInterval)
+		listOption := configInformers.WithTweakListOptions(func(opt *metav1.ListOptions) {
+			opt.FieldSelector = fields.OneTermEqualSelector(metav1.ObjectNameField, meshConfigName).String()
```

Comment:
@nojnhuh Did we try to use a field selector for the MeshConfig in the past? I thought we decided against it for some reason/bug

Reply:
FWIW, copied the field selector from the [configurator client](https://github.com/openservicemesh/osm/blob/v1.1.1/pkg/configurator/client.go#L30) in v1.1.1

Reply:
I vaguely remember something coming up about field selectors, but I don't think it was this. It worked here as expected as far as I remember.